### DISCUSSION TITLE
喜乐你好👋，我想请教下后续cs50视频会更新么？

### DISCUSSION THREAD

**Discussion by SicOverflow** [Xilesun/CS50x#14] [Q&A] [answered]
### 你的运行环境是?

cs50.dev

### 请详细描述你的问题，操作步骤。并附上完整的报错信息截图，或其他有效信息。

如果实在抽不出时间能在code里放点提示性的批注么？被week2的加密算法弄懵了，不知道怎么构建函数框架🥹
并且十分感谢您的往期视频！！！真的！真的！给了我很大帮助，感激不尽！👍

**Comment by 2013xile** [comment]
是哪一个题目？week2 作业仓库里是有参考代码的，可以具体说说哪个地方没看懂。

**Reply by SicOverflow** [reply]
[https://github.com/2013xile/CS50x/blob/main/code/probs/prob2/substitution/substitution.c](url)

是Pro2里面的substitution，用到了很多不了解的函数
<img width="442" alt="截屏2024-11-18 10 03 20" src="https://github.com/user-attachments/assets/c568e4f9-aade-46a2-a789-4610707548e2">

例如头次知道int main(int argc, string argv[])，并且大略的看了下程序里“参数”能发挥什么样的作用等。
其实最主要的问题感觉还是函数库里的了解的不全，导致不知道该怎么解决给的相应条件。还有不太懂“！”
是什么有什么用。
<img width="687" alt="截屏2024-11-18 10 32 53" src="https://github.com/user-attachments/assets/7729c6f9-e113-4b50-b64d-bb506a9d1226">

int count [26] = {0}；又是什么意思😧
问题有点多...（输入于水课中）

**Comment by 2013xile** [comment, marked as answer]
参考一下这期视频 https://www.bilibili.com/video/BV1PB4y1o7Ey
数组初始化和命令行参数部分
函数都可以参考 https://manual.cs50.io/

`!=` 表示不等于
`!` 表示否——不是 xxx 条件，例如 `!isalpha(c)` 表示 c 不是字母。

**Reply by SicOverflow** [reply]
感谢，光顾着闭门造车了🥸

**Reply by SicOverflow** [reply]
对了xile，makefile和Cmake是一个东西么？之前捣鼓Windows上的wsl装lib-cs50库的时候记得装完辅助轮和函数库后就在文档树哪里创建了makefile文件，但由于当时啥也不懂就跟着做了，后来了解了些许gcc和链接库等知识，知道了这俩东西在后续也挺重要的，特有此问🧐

**Reply by 2013xile** [reply]
这类问题可以直接查一下，比如 https://juejin.cn/post/6854573222017368078

**Reply by SicOverflow** [reply]
好的🫡
